torch/csrc/profiler/util.h

```diff
@@ -35,6 +35,24 @@
 #endif
 #endif
 
+// Define SOFT_ASSERT
+#ifdef NDEBUG
```

Comment:
As written, `NDEBUG` takes priority over the environment variable. I think it should be the other way around. (And of course that means the environment variable must be able to both force enable and force disable.) My reasoning is that I want to add unit test for the profiler that test various kinds of pathological conditions / invalid inputs and assert that they are handled gracefully. However if the behavior depends on the build flag then it's impossible to write unit tests that will always work.

Reply:
Maybe let the env var disable the HANDLE_ERROR in both cases?

Reply:
Ah I see what you are saying.